R/assocTestSingle.R

```diff
@@ -45,14 +47,23 @@ setMethod("assocTestSingle",
                       geno <- imputedDosage(gdsobj, use.names=FALSE)[sample.index,,drop=FALSE]
                   }
 
-                  # take note of number of non-missing samples
+                   # take note of number of non-missing samples
                   #n.obs <- colSums(!is.na(geno))
                   n.obs <- .countNonMissing(geno, MARGIN = 2)
 
+                  if(geno.test=='recessive'){ # is recessive
+                      ## if wanting to test a recessive model, the genotypes 0 and 1 are '0' and the genotype 2 is '1',
+                      ##eg. indicator of participant having 2 copies of the minor allele
+                      geno[geno==1] <- 0
```

Comment:
Need to allow for missing values in the `geno` vector. I think using `%in%` instead of `==` will fix this.

Reply:
Turns out we should use `==` after all, as `%in%` doesn't work for `Matrix` objects.